### ISSUE TITLE
Travic CI fails with message BUILDROOT: No such file or directory

### ISSUE THREAD

**Issue by FrankR85** [mmitch/cobol-travis#12]
I try to build [this repo](https://github.com/FrankR85/ToDoListe) with Travis and get the following error:
```
$ sudo make -C <BUILDROOT> install-gnucobol
/home/travis/.travis/functions: line 104: BUILDROOT: No such file or directory
The command "sudo make -C <BUILDROOT> install-gnucobol" failed and exited with 1 during .
```
However, I think I've set the ```BUILDPATH``` correctly [here](https://github.com/FrankR85/ToDoListe/blob/master/Makefile#L1). Locally, the ```make``` works.

**Comment by mmitch** [comment]
`BUILDPATH` is a variable inside the main makefile, while `<BUILDROOT>` is a placeholder that has to be explicitely edited by you because `.travis.yml` does not read your main Makefile.

Also: `PATH` != `ROOT` :wink:

It's documented like this:

> 5. If you want [TravisCI integration](https://travis-ci.org), copy
>   `template/.travis.yml` to the root directory of your project and
>   replace `<BUILDROOT>` by the directory of _cobol-build_.

Any ideas of how to make this more prominent?

The reason for this is that I don't want to mess up the main Makefile with another dispatched target that is only useful if TravisCI is used.

I could put `cobol-build` into the template so it will work out-of-the-box unless the user has put _cobol-travis_ into a different directory.

**Comment by FrankR85** [comment]
Ah, ok. Sorry, I didn't read the instructions thoroughly enough.

I suggest just putting ```cobol-build``` into the template and wait for the first user to modify this himself. :1st_place_medal:

**Comment by mmitch** [comment]
Documentation and template updated with commit 4e5f3d1
Care for a short review?

**Comment by FrankR85** [comment]
Short sidenote, I have to dig deeper to understand the reasons, but I can't update my submodule with
```
git submodule update cobol-build
```
Nothing happens then.
I have to use:
```
git submodule update --remote --merge
```

**Comment by FrankR85** [comment]
Yes, the documentation is much clearer now and it should work out of the box!

**Comment by mmitch** [comment]
keeping this open to check out the proper submodule commands

**Comment by mmitch** [comment]
The submodule update instructions were indeed wrong, I fixed that with commit ff2be29.

* `git submodule update` does a checkout of the submodule at the commit that is stored in your project
* `git submodule update --remote` does a checkout of the submodule at the most current commit of the remote repository

So to update your repository with the latest version of _cobol-build_, you need to use `--remote`.  Then you should commit your new submodule state.

Anybody who uses your repository can then update without `--remote` to get to the state that you just checked in.  (And if you roll back your repository to an earlier state, you should also do a submodule update without `--remote` to roll the submodule back as well so it matches the rest of your repository at that point in time.)

**Comment by mmitch** [comment]
btw: `--merge` would only be relevant if you did any changes and commits in the submodule on your end
